lib/src/WebSocketConnectionImpl.cc

```diff
@@ -344,14 +350,17 @@ bool WebSocketMessageParser::parse(trantor::MsgBuffer *buffer)
             }
             else if (indexFirstMask == 10)
             {
-                length = (unsigned char)(*buffer)[2];
-                length = (length << 8) + (unsigned char)(*buffer)[3];
-                length = (length << 8) + (unsigned char)(*buffer)[4];
-                length = (length << 8) + (unsigned char)(*buffer)[5];
-                length = (length << 8) + (unsigned char)(*buffer)[6];
-                length = (length << 8) + (unsigned char)(*buffer)[7];
-                length = (length << 8) + (unsigned char)(*buffer)[8];
-                length = (length << 8) + (unsigned char)(*buffer)[9];
+                length = 0;
+                for (int i = 2; i <= 9; ++i)
+                {
+                    if (length > ((std::numeric_limits<size_t>::max)() >> 8))
```

Comment:
This does not feel right. `size_t` is different on 32 and 64bit systems. So the max size will be 24 and 48bits respectively. Do you intend to use `uint64_t`?

Reply:
The judgment is followed by an 8-bit left shift operation, so the maximum values ​​are 32-bit and 64-bit respectively. I think this judgment protects the 32-bit system from overflow.